apis/fluent/atrium-api-fluent/src/commonTest/kotlin/ch/tutteli/atrium/api/fluent/en_GB/CharSequenceToContainAtLeastElementsOfExpectationsTest.kt

```diff
@@ -0,0 +1,149 @@
+            atLeastIgnoringCaseDescr to ("$toContain.$ignoringCase.$atLeast.$elementsOf" to Companion::toContainAtLeastIgnoringCaseElementsOf)
+
+        private fun toContainAtLeastIgnoringCaseElementsOf(
+            expect: Expect<CharSequence>,
+            atLeast: Int,
+            a: Any,
+            aX: Array<out Any>
+        ): Expect<CharSequence> =
+            if (aX.isEmpty()) {
+                if (atLeast == 1) expect.toContain.ignoringCase.elementsOf(listOf(a))
+                else expect.toContain.ignoringCase.atLeast(atLeast).elementsOf(listOf(a))
+            } else {
+                if (atLeast == 1) expect.toContain.ignoringCase.elementsOf(listOf(a, *aX))
+                else expect.toContain.ignoringCase.atLeast(atLeast).elementsOf(listOf(a, *aX))
+            }
+
+        private val atLeastButAtMostDescr = { what: String, timesAtLeast: String, timesAtMost: String ->
+            "$toContain $what $atLeast $timesAtLeast $butAtMost $timesAtMost"
+        }
+
+        private fun getAtLeastButAtMostElementsOfTriple() =
+            atLeastButAtMostDescr to ("$toContain.$atLeast.$butAtMost.$elementsOf" to Companion::toContainAtLeastButAtMostElementsOf)
+
+        private fun toContainAtLeastButAtMostElementsOf(
+            expect: Expect<CharSequence>,
+            atLeast: Int,
+            butAtMost: Int,
+            a: Any,
+            aX: Array<out Any>
+        ) = expect.toContain.atLeast(atLeast).butAtMost(butAtMost).elementsOf(listOf(a, *aX))
+
+        private val atLeastButAtMostIgnoringCaseDescr = { what: String, timesAtLeast: String, timesAtMost: String ->
+            "$toContain $ignoringCase $what $atLeast $timesAtLeast $butAtMost $timesAtMost"
+        }
+
+        private fun getAtLeastButAtMostIgnoringCaseElementsOfTriple() =
+            atLeastButAtMostIgnoringCaseDescr to ("$toContain.$ignoringCase.$atLeast.$butAtMost.$elementsOf" to Companion::toContainAtLeastButAtMostIgnoringCaseElementsOf)
+
+        private fun toContainAtLeastButAtMostIgnoringCaseElementsOf(
+            expect: Expect<CharSequence>,
+            atLeast: Int,
+            butAtMost: Int,
+            a: Any,
+            aX: Array<out Any>
+        ) = expect.toContain.ignoringCase.atLeast(atLeast).butAtMost(butAtMost).elementsOf(listOf(a, *aX))
+
+        private fun getNotToContainPair() = toContainNot to ::getErrorMsgNotToContain
+
+        private fun getErrorMsgNotToContain(times: Int) = "use $toContainNot instead of $atLeast($times)"
+
+        private fun getExactlyPair() = exactly to Companion::getErrorMsgExactly
+
+        private fun getErrorMsgExactly(times: Int) =
+            "use $exactly($times) instead of $atLeast($times).$butAtMost($times)"
+
+        internal fun getErrorMsgAtLeastButAtMost(timesAtLeast: Int, timesButAtMost: Int) =
+            "specifying $butAtMost($timesButAtMost) does not make sense if $atLeast($timesAtLeast) was used before"
+
+        private fun <T> iterableLikeToIterableTestFactory(
```

Comment:
you should move this function to atrium-specs so that we can re-use it at several places. In your PR here and in the infix api but there are more places to come where we want to use it in the future (basically where ch.tutteli.atrium.specs.integration.IterableLikeToIterableSpec) is used now. Put it into a new file in:
`misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/utils/iterableLikeToIterableTestFactory.kt`